packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -859,19 +916,108 @@ function useEffectImpl(fiberEffectTag, hookEffectTag, create, deps): void {
     }
   }
 
-  currentlyRenderingFiber.effectTag |= fiberEffectTag;
+  sideEffectTag |= fiberEffectTag;
   workInProgressHook.memoizedState = pushEffect(
     hookEffectTag,
     create,
     destroy,
     nextDeps,
   );
+}
+
+function mountEffect(
+  create: () => mixed,
+  deps: Array<mixed> | void | null,
+): void {
   if (__DEV__) {
-    currentHookNameInDev = null;
+    currentHookNameInDev = 'useEffect';
+  }
+  const hook = mountWorkInProgressHook();
+  return mountEffectImpl(
+    hook,
+    UpdateEffect | PassiveEffect,
+    UnmountPassive | MountPassive,
+    create,
+    deps,
+  );
+}
+
+function updateEffect(
+  create: () => mixed,
+  deps: Array<mixed> | void | null,
+): void {
+  if (__DEV__) {
+    currentHookNameInDev = 'useEffect';
   }
+  const hook = updateWorkInProgressHook();
```

Comment:
Same thing as in the mount case. Let's put it in `updateEffectImpl()` so it's just one function call in here. (Later we can play around with bound functions.)